lib/segment/src/payload_storage/condition_checker.rs

```diff
@@ -140,18 +137,34 @@ impl ValueChecker for Match {
     }
 }
 
-impl ValueChecker for Range<FloatPayloadType> {
+impl ValueChecker for RangeInterface {
     fn check_match(&self, payload: &Value) -> bool {
-        match payload {
-            Value::Number(num) => num
-                .as_f64()
-                .map(|number| self.check_range(number))
-                .unwrap_or(false),
-            _ => false,
+        match self {
+            RangeInterface::Float(condition) => condition.check_match(payload),
+            RangeInterface::Int(condition) => condition.check_match(payload),
+            RangeInterface::DateTime(condition) => condition.check_match(payload),
         }
     }
 }
 
+impl ValueChecker for Range<FloatPayloadType> {
+    fn check_match(&self, payload: &Value) -> bool {
+        payload
+            .as_number()
+            .and_then(|x| x.as_f64().or_else(|| x.as_i64().map(|x| x as f64)))
+            .map_or(false, |x| self.check_range(x))
+    }
+}
+
+impl ValueChecker for Range<IntPayloadType> {
+    fn check_match(&self, payload: &Value) -> bool {
+        payload
+            .as_number()
+            .and_then(|x| x.as_i64().or_else(|| x.as_f64().map(|x| x as i64)))
+            .map_or(false, |x| self.check_range(x))
+    }
+}
+
```

Comment:
For unindexed fields, we now try to convert into the type of the range to do the checks